### ISSUE TITLE
Proposal: "HPC Workshops" organization to host materials indefinitely

### ISSUE THREAD

**Issue by tkphd** [hpc-carpentry/coordination#71]
Learners want to have permanent access to lesson materials, but with unclear and non-uniform institutional GitHub retention policies, it is hard at present to promise any sort of longevity. 

I therefore propose that we provide a central place for lesson and workshop forks to be collected, in order to allay this uncertainty and therefore encourage workshop hosts to put the effort in to customizing the lesson template for the specific venue of the workshop. 

To limit clutter in the HPC Carpentry organization, I further propose creating a separate GitHub organization, "HPC Workshops," to host these customized repositories, with ownership permissions granted to the Steering Committee and write permissions for each repository granted to the workshop coordinator and any additional accounts they nominate.

**Comment by tkphd** [comment]
*Temperature check:* Please respond to this comment with a "thumbs up", "thumbs down", or "eyes" to register your opinion in favor, in opposition, or indifferent to the Proposal. This is not a vote *per se*, but will inform the discussion of the Proposal at our next meeting.

**Comment by wirawan0** [comment]
This sounds like a nice idea, which may free HPC sites from having to host their own versions of the lessons.

But have you thought about potential challenges with this approach?

* Lesson providers may update the lesson from one year to another. What will happen with the different versions of the lessons? Do learners have to yield to the newest version of the lesson at all times?
* Do you archive the lesson _after_ the workshop has been conducted? Or do we let "HPC Workshops" to host the customized lesson from the get to?

Other Carpentry or Carpentry-like communities: Any of them doing this? I guess because we almost can't make the lesson useful for our own uses without customizing, while others don't have to do this because Bash shell is the same everywhere.

**Comment by tkphd** [comment]
When a workshop fork is created, it is named using a standard convention such as **YYYY-MM-DD-institution-topic**, e.g., 
"2021-08-25-nist-hpc-shell". Changes are made up to, including, and possibly after the workshop ends, reflecting feedback from learners. GitHub sorts repos by modification date, most recent first, so folks could browse the latest workshops easily, or search for older ones.

**Comment by tkphd** [comment]
All righty, some great feedback from @tobyhodges: why one repo per institution per day? Wouldn't folks who change the site template once want to just reuse the repo for later workshops? --> Absolutely!

For each workshop, we can generate one repository per lesson, per institution, and per cluster, representing a properly templated build. These would be maintained as the core lesson materials change. Since we intend to maintain them, the repos could/should be created under HPC Carpentries, without much more administrative overhead. E.g.,

* hpc-carpentries/**brac-slurm-hpc-intro**
* hpc-carpentries/**brac-slurm-hpc-parallel-novice**
* hpc-carpentries/**brac-slurm-hpc-python**
* hpc-carpentries/**nist-ctcms-hpc-intro**
* hpc-carpentries/**nist-raritan-hpc-intro**

**Comment by annajiat** [comment]
would it be possible to add "u" after "brac"?

**Comment by tkphd** [comment]
Absolutely! The list is for example, only; actual forks should be created by, or in close collaboration/communication with, the workshop coordinator (i.e., yourself). If you would like to do so, please feel free to create the forks you need under this organization, and we will keep them around for as long as we can ("indefinitely," which may or may not evaluate the same as "in perpetuity"). I would request a naming scheme like `institution-cluster-hpc-intro` to help everyone keep them straight, but the particulars are up to you.

**Comment by tkphd** [comment]
Following this morning's positive feedback, as well as the support & creative streamlining offered by Toby last coworking hour, I went ahead and created the proposed organization:

[hpc-workshops](https://github.com/hpc-workshops)

I've invited most of the frequent participants I could recall off the top of my head to join. If you didn't receive a notification, please let me know! The purpose of this new org is to store, with some reasonable persistence, forks of hpc-carpentry repositories customized for specific sites using the snippet library. These tuned lessons can then be linked directly from workshop outlines, and reused, so that learners have reference materials to look back on.
